Musl does not have rand_r() API

Mark it to be glibc specific

Upstream-Status: Pending
Signed-off-by: Khem Raj <raj.khem@gmail.com>

--- a/third_party/fontconfig/include/config.h
+++ b/third_party/fontconfig/include/config.h
@@ -157,10 +157,13 @@
 #define HAVE_RANDOM 1
 
 /* Define to 1 if you have the `random_r' function. */
-#define HAVE_RANDOM_R 1
-
+#ifdef __GLIBC__
+# define HAVE_RANDOM_R 1
+#endif
 /* Define to 1 if you have the `rand_r' function. */
-#define HAVE_RAND_R 1
+#ifdef __GLIBC__
+# define HAVE_RAND_R 1
+#endif
 
 /* Define to 1 if you have the `readlink' function. */
 #define HAVE_READLINK 1
